5calls/app/src/androidTest/java/org/a5calls/android/a5calls/controller/MainActivityHappyPathTest.java

```diff
@@ -0,0 +1,406 @@
+
+    private MockHttpStack mHttpStack;
+    private RequestQueue mOriginalRequestQueue;
+    private FiveCallsApi mOriginalApi;
+    private String mOriginalAddress;
+
+    /**
+     * A simple IdlingResource that waits for a condition to be true.
+     */
+    private static class ConditionIdlingResource implements IdlingResource {
+        private final AtomicBoolean mIsIdle = new AtomicBoolean(false);
+        private ResourceCallback mResourceCallback;
+
+        public void setIdle(boolean isIdle) {
+            mIsIdle.set(isIdle);
+            if (isIdle && mResourceCallback != null) {
+                mResourceCallback.onTransitionToIdle();
+            }
+        }
+
+        @Override
+        public String getName() {
+            return ConditionIdlingResource.class.getName();
+        }
+
+        @Override
+        public boolean isIdleNow() {
+            return mIsIdle.get();
+        }
+
+        @Override
+        public void registerIdleTransitionCallback(ResourceCallback callback) {
+            mResourceCallback = callback;
+        }
+    }
+
+    @Before
+    public void setUp() {
+        // Save original state
+        mOriginalRequestQueue = AppSingleton.getInstance(
+                InstrumentationRegistry.getInstrumentation().getTargetContext()).getRequestQueue();
+        mOriginalApi = AppSingleton.getInstance(
+                InstrumentationRegistry.getInstrumentation().getTargetContext()).getJsonController();
+
+        // Save original location
+        mOriginalAddress = AccountManager.Instance.getAddress(
+                InstrumentationRegistry.getInstrumentation().getTargetContext());
+
+        // Set a mock location to avoid location prompts
+        AccountManager.Instance.setAddress(
+                InstrumentationRegistry.getInstrumentation().getTargetContext(),
+                "90210");
+
+        // Mark tutorial as seen to bypass onboarding screen
+        AccountManager.Instance.setTutorialSeen(
+                InstrumentationRegistry.getInstrumentation().getTargetContext(),
+                true);
+
+        // Create mock HTTP stack
+        mHttpStack = new MockHttpStack();
+    }
+
+    @After
+    public void tearDown() {
+        // Restore original state
+        AppSingleton.getInstance(
+                InstrumentationRegistry.getInstrumentation().getTargetContext())
+                .setRequestQueue(mOriginalRequestQueue);
+        AppSingleton.getInstance(
+                InstrumentationRegistry.getInstrumentation().getTargetContext())
+                .setFiveCallsApi(mOriginalApi);
+
+        // Restore original location
+        AccountManager.Instance.setAddress(
+                InstrumentationRegistry.getInstrumentation().getTargetContext(),
+                mOriginalAddress);
+    }
+
+    @Test
+    public void testMainUILoadsCorrectly() throws JSONException {
+        // Create mock issues response
+        JSONArray issuesArray = new JSONArray();
```

Comment:
FYI there's a real issues JSON in FakeJSONData.java. There's another FakeJSONData.java that has `public static JSONArray GetIssueJSON`, too.

I suggest either moving this to a helper in FakeJSONData (`getSimpleIssuesJSON()`) or using the existing json string there, rather than building it up within the test.

Reply:
now using the existing FakeJSONData